codec-http/src/main/java/io/netty/handler/codec/http/cookie/ServerCookieDecoder.java

```diff
@@ -75,7 +75,7 @@ private ServerCookieDecoder(boolean strict) {
         boolean rfc2965Style = false;
         if (header.regionMatches(true, 0, RFC2965_VERSION, 0, RFC2965_VERSION.length())) {
             // RFC 2965 style cookie, move to after version value
-            i = header.indexOf(';') + 1;
+            i = getDelimiterPos(header, 0) + 1;
```

Comment:
getDelimiterPos is just like indexOf, second parameter is an offset, could you expand what you mean in your first question?
Second question coment: the implementation is doing stuff like before expanded by adding comma as a delimiter. getDelimiterPos returns -1 in case ;, is not found.